espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/IntentHelpers.kt

```diff
@@ -0,0 +1,298 @@
+                intent.putExtra(k, v)
+            },
+            "eia" to fun(key, value) {
+                val (k, v) = requirePair(String::class, key, value)
+                val numbers = v.split(",")
+                        .map {
+                            try {
+                                it.trim().toInt()
+                            } catch (e: NumberFormatException) {
+                                throw IllegalArgumentException(
+                                        "The value is expected to be a comma-separated list of integers. " +
+                                                "'$v' is given instead", e)
+                            }
+                        }
+                        .toIntArray()
+                intent.putExtra(k, numbers)
+            },
+            "ela" to fun(key, value) {
+                val (k, v) = requirePair(String::class, key, value)
+                val numbers = v.split(",")
+                        .map {
+                            try {
+                                it.trim().toLong()
+                            } catch (e: NumberFormatException) {
+                                throw IllegalArgumentException(
+                                        "The value is expected to be a comma-separated list of long integers. " +
+                                                "'$v' is given instead", e)
+                            }
+                        }
+                        .toLongArray()
+                intent.putExtra(k, numbers)
+            },
+            "efa" to fun(key, value) {
+                val (k, v) = requirePair(String::class, key, value)
+                val numbers = v.split(",")
+                        .map {
+                            try {
+                                it.trim().toFloat()
+                            } catch (e: NumberFormatException) {
+                                throw IllegalArgumentException(
+                                        "The value is expected to be a comma-separated list of float numbers. " +
+                                                "'$v' is given instead", e)
+                            }
+                        }
+                        .toFloatArray()
+                intent.putExtra(k, numbers)
+            },
+            "ecn" to fun(key, value) {
+                val (k, name) = requirePair(String::class, key, value)
+                intent.putExtra(k, name.toComponentName())
+            }
+    )
+
+    for ((optName, optValue) in options) {
+        val handler = handlersMapping[optName]
+                ?: throw IllegalArgumentException("The option named '$optName' is not known. " +
+                        "Only the following options are supported: ${handlersMapping.keys}")
+        handler(optName, optValue)
+    }
+
+    intent.setDataAndType(data, type)
+
+    require(hasIntentInfo) { "Either intent action, data, type or category must be supplied" }
```

Comment:
I got a chance to test this.
1. Currently, to succeed, all ''action', 'category', 'component' and 'flags' are needed. Therefore, we should throw this error if any of these is missing
2. OR, It will be good to have default values as
'action' => 'android.intent.action.MAIN',
'category' => 'android.intent.category.LAUNCHER',
'flags' => 'FLAG_ACTIVITY_NEW_TASK'
'category' => fullyQualifiedAppActivity()
3. We can report a specific cause of failure. eg if user missed specifying FLAG_ACTIVITY_NEW_TASK we could report failure cause as
"android.util.AndroidRuntimeException: Calling startActivity() from outside of an Activity  context requires the FLAG_ACTIVITY_NEW_TASK flag. Is this really what you want?" instead of "java.lang.RuntimeException: Could not launch activity."

Reply:
Thanks, I'll add the necessary fixes

Reply:
I was analysing the preconditions above and for now I don't think this all is necessary to be implemented in the code. The intent options capability itself is only going to be used by advanced engineers. So it would be fine to just properly describe the feature in the documentation. 
I would be very grateful if you could write such documentation with examples.

Reply:
Ok, makes sense. What is the place to write documentation?

Reply:
https://github.com/appium/appium/tree/master/docs/en/writing-running-appium/android/